File: Napakalaki/src/NapakalakiGame/Treasure.java
package NapakalakiGame;

/**
 *
 * @author carlos
 */
public class Treasure {
    private String name;
    private int bonus;
    private TreasureKind treasure;
    
    
    Treasure(String n, int bonus1, TreasureKind t){
        name=n;
        bonus=bonus1;
        treasure=t;
    }
    
    
    
    public String getName(){
        return name;
    }
    
    public int getBonus(){
        return bonus;
    }
    
    public TreasureKind getType(){
        return treasure;
    }
    
    @Override
    public String toString() {
        return "Nombre: " + this.name + ". Tipo: " + this.treasure;
    }
}

File: Napakalaki-Ruby/lib/Treasure.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module NapakalakiGame
  
class Treasure
  def initialize( n, bonus1, t)
    @name=n
    @bonus=bonus1
    @type=t
  end
    
    attr_reader:name
    attr_reader:bonus
    attr_reader:type    
    
  def to_s
  "#{@name}. Tipo #{@type}"
  end
    
end

end